File: server/models/Post.js
const mongoose = require("mongoose");

const postSchema = new mongoose.Schema(
  {
    title: { type: String },
    body: { type: String },
    likes: [{ type: String }],
    array_tags: [{ type: String, ref: "Tag" }],
    author_username: { type: String, ref: "User" },
  },
  { timestamps: true },
);

module.exports = mongoose.model("Post", postSchema);

File: mern_mobile_app/lib/models/post.dart
import 'package:chip_in/models/tag.dart';

class Post {
  final String id;
  final String title;
  final String body;
  final List<String> likes;
  final List<Tag> tags;
  final String authorUsername;
  final DateTime datePosted;

  Post({
    required this.id,
    required this.title,
    required this.body,
    required this.likes,
    required this.tags,
    required this.authorUsername,
    required this.datePosted
  });
}